pkg/bindings/images/build.go

```diff
@@ -607,7 +608,26 @@ func Build(ctx context.Context, containerFiles []string, options types.BuildOpti
 
 							modifiedSrc := fmt.Sprintf("src=%s", filepath.Base(tmpSecretFile.Name()))
 							modifiedOpt = append(modifiedOpt, modifiedSrc)
-						} else {
+						case "env":
+							// read specified env var into a tmp file
+							// move tmp file to tar and change secret source to relative tmp file
+							data := []byte(os.Getenv(val))
```

Comment:
This should be a os.LookupEnv() and then throw an error if the environment value is not set.